internal/otelcollector/config/log/agent/testdata/config.yaml

```diff
@@ -92,6 +92,26 @@ receivers:
               type: severity_parser
               if: attributes.level != nil
               parse_from: attributes.level
+            - id: trace-parent-parser
+              type: regex_parser
+              if: attributes.trace_id == nil
+              parse_from: attributes.traceparent
+              regex: ^[0-9a-f]{2}-(?P<trace_id>[0-9a-f]{32})-(?P<span_id>[0-9a-f]{16})-(?P<trace_flags>[0-9a-f]{2})$
+              trace:
+                trace_id:
+                    parse_from: attributes.trace_id
+                span_id:
+                    parse_from: attributes.span_id
+                trace_flags:
+                    parse_from: attributes.trace_flags
+            - id: trace-parser
```

Comment:
Am not sure about the order. If someone provides as log attributes "trace_id" AND "traceparent", who should win? I vote for trace_id

Reply:
I added a new router operator to route to the correct parser, the `trace-parser` has precedence over the `trace-parent-parser`, and `traceparent` is parsed when no `trace_id` is present.

After successfully parsing the new attributes removed, in the case `trace-parser` the attributes `trace_id`, `span_id`, and `trace_flags` were removed, the `trace-parent-parser` removes in addition to `trace-parser` also `traceparent` attribute